packages/atlas/src/views/studio/PlaylistWorkspace/VideoSelectorDialog/VideoSelectorDialog.tsx

```diff
@@ -0,0 +1,118 @@
+import { Text } from '@/components/Text'
+import { Button } from '@/components/_buttons/Button'
+import { Input } from '@/components/_inputs/Input'
+import { DialogModal } from '@/components/_overlays/DialogModal'
+import { VideoListItem, VideoListItemLoader } from '@/components/_video/VideoListItem'
+import { useDebounceValue } from '@/hooks/useDebounceValue'
+import { useAuthorizedUser } from '@/providers/user/user.hooks'
+import { SentryLogger } from '@/utils/logs'
+
+import { Wrapper } from './VideoSelectorDialog.styles'
+
+type VideoSelectorDialogProps = {
+  onHide: () => void
+  show: boolean
+  onSelect: (arg: string[]) => void
+  initialSelected?: string[]
+}
+const INITIAL_FIRST = 50
+
+export const VideoSelectorDialog = ({ onHide, show, onSelect, initialSelected = [] }: VideoSelectorDialogProps) => {
+  const [selectedVideos, setSelectedVideos] = useState<string[]>(initialSelected)
+  const [search, setSearch] = useState<string>('')
+  const { channelId } = useAuthorizedUser()
+  const debouncedSearch = useDebounceValue(search, 200)
+  const { edges, loading } = useFullVideosConnection(
+    {
+      first: INITIAL_FIRST,
+      orderBy: VideoOrderByInput.CreatedAtDesc,
+      where: {
+        channel: {
+          id_eq: channelId,
+        },
+        title_contains: debouncedSearch,
+      },
+    },
+    {
+      notifyOnNetworkStatusChange: true,
+      onError: (error) => SentryLogger.error('Failed to fetch videos', 'MyVideosView', error),
+    }
+  )
+
+  useEffect(() => {
+    if (show) {
+      setSelectedVideos(initialSelected)
+    }
+    // need to update selected videos when modal opens
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [show])
+
+  return (
+    <DialogModal
+      show={show}
+      size="medium"
+      primaryButton={{
+        text: 'Add videos',
+        onClick: () => {
+          onSelect(selectedVideos)
+          onHide()
+        },
+      }}
+      secondaryButton={{ text: 'Cancel', onClick: onHide }}
+      additionalActionsNode={
+        <>
+          <Button
+            variant="tertiary"
+            onClick={() => edges?.length && setSelectedVideos(edges.map((edge) => edge.node.id))}
+          >
+            Select all
+          </Button>
+          {!!selectedVideos.length && (
+            <Text as="p" variant="t300" color="colorTextMuted">
+              {selectedVideos.length} video(s) selected
+            </Text>
+          )}
+        </>
+      }
+      actionDivider
+    >
+      <Wrapper>
+        <Input
+          value={search}
+          onChange={(e) => setSearch(e.target.value)}
+          nodeStart={<SvgActionSearch />}
+          placeholder="Search videos to add"
+        />
+        {loading ? (
+          <div style={{ width: '100%' }}>
+            {Array.from({ length: 3 }, (_, idx) => (
+              <VideoListItemLoader key={idx} variant="small" />
+            ))}
+          </div>
+        ) : (
+          <div>
```

Comment:
Not sure, because the design doesn't specify this well, but I think the list should have its own scrollbar. The search box should be always visible and placed on top when you scroll. Right now this is not a pleasant UX IMHO: 
![image](https://user-images.githubusercontent.com/51168865/210768018-0bc1193a-2284-49fb-8396-9280994bb978.png)


I would consult this with a designer. This design could suggest that it should work like that(look at the last item):  https://www.figma.com/file/XfKZ5ZSavt88rVfiKf0tBv/My-playlists-page-%26-Playlist-workspace?node-id=905%3A9397&t=WyGuCWg3Z3ZFL3mB-4